packages/core/src/check-license-status.ts

```
@@ -0,0 +1,6 @@
+export type LicenseStatus = "allowed" | "disallowed" | "unknown";
```

any different classification ideas/preferences?

---

What if we consider licenses as rules and get inspired by the severity of ESLint/Biome?

https://eslint.org/docs/latest/use/configure/rules#rule-severities

---

- Are `needsReview` cases possible? In case a license doesn't fit any of our checks and actually needs manual review?
- According to this list: https://spdx.org/licenses/ we could also mark the licenses as `deprecated`.

---

> What if we consider licenses as rules and get inspired by the severity of ESLint/Biome?

@benedyktdryl good idea. So if I understand correctly would the user configure what to do with each case? Like:
```
"unknown": "off"/"warn"/"error",
"blacklist": "off"/"warn"/"error"
```
Does it make sense to make `whitelist`ed packages configurable as such though? Should it be skipped? I don't really see a scenario where somebody would want to throw an error or a warning when a license passes the check.

---

@matt-jb @F-Kublin 

> Are needsReview cases possible? In case a license doesn't fit any of our checks and actually needs manual review?

It will be the `unknown` case. I like the idea of reviewing such packages.

I think it would be nice to add an `override` field to the package's config so if the run returns `unknown` packages the user will be able to manually set up whether the package should pass, log a warning or throw an error.

> Are needsReview cases possible? In case a license doesn't fit any of our checks and actually needs manual review?
According to this list: https://spdx.org/licenses/ we could also mark the licenses as deprecated.

Valid point

> Would it be better for a deprecated license to be an automatic error or to mark it as deprecated, but still let it pass the check?

I like the idea of adding an additional `deprecated` field and letting the license pass
Lots of licenses use deprecated license field formats or deprecated licenses unfortunately